libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/DirectionsSession.kt

```diff
@@ -16,6 +16,12 @@ internal interface DirectionsSession {
      */
     val routes: List<DirectionsRoute>
 
+    /**
+     * Routes update reason. Initially [RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP]
+     */
+    @RoutesExtra.RoutesUpdateReason
+    val routesUpdateReason: String
```

Comment:
Looks like we don't need to add this to the interface, it can be private in `MapboxDirectionsSession`.